via/views/proxy.py

```diff
@@ -10,6 +10,8 @@ def proxy(request):
     # Strip leading '/' and normalize URL.
     url = url_from_user_input(request.path[1:])
 
+    request.checkmate.raise_if_blocked(url)
```

Comment:
Were these ones not checkmate'd at all?

Reply:
No. I'm guessing the theory was that we were "only" peeking at the data to see what it was / that we would check it properly when it got to the next step, but the truth is we can attempt to make a quick query with HEAD and the like, but we're not in control. The server can take as long as it likes.

In the case of the Wayback machine problems we've been having, the request is the message, the caller doesn't care about the content as they are trying to trigger it to download a website. So just issuing the call puts us on the hook for 30 seconds or something.